src/raw.rs

```diff
@@ -445,30 +456,56 @@ pub fn replicate_verbatim(ctx: *mut RedisModuleCtx) -> Status {
     unsafe { RedisModule_ReplicateVerbatim.unwrap()(ctx).into() }
 }
 
+fn load<T>(
+    rdb: *mut RedisModuleIO,
+    f: Option<unsafe extern "C" fn(io: *mut RedisModuleIO) -> T>,
+) -> Result<T, Error> {
+    let res = unsafe { f.unwrap()(rdb) };
+    if is_io_error(rdb) {
+        Err(RedisError::short_read().into())
+    } else {
+        Ok(res)
+    }
+}
+
+fn load_with_param<T, V>(
+    rdb: *mut RedisModuleIO,
+    f: Option<unsafe extern "C" fn(io: *mut RedisModuleIO, *mut V) -> T>,
+    v: *mut V,
+) -> Result<T, Error> {
+    let res = unsafe { f.unwrap()(rdb, v) };
+    if is_io_error(rdb) {
+        Err(RedisError::short_read().into())
+    } else {
+        Ok(res)
+    }
+}
+
 #[allow(clippy::not_unsafe_ptr_arg_deref)]
-pub fn load_unsigned(rdb: *mut RedisModuleIO) -> u64 {
-    unsafe { RedisModule_LoadUnsigned.unwrap()(rdb) }
+pub fn load_unsigned(rdb: *mut RedisModuleIO) -> Result<u64, Error> {
+    unsafe { load::<u64>(rdb, RedisModule_LoadUnsigned) }
 }
 
 #[allow(clippy::not_unsafe_ptr_arg_deref)]
-pub fn load_signed(rdb: *mut RedisModuleIO) -> i64 {
-    unsafe { RedisModule_LoadSigned.unwrap()(rdb) }
+pub fn load_signed(rdb: *mut RedisModuleIO) -> Result<i64, Error> {
+    unsafe { load::<i64>(rdb, RedisModule_LoadSigned) }
 }
 
 #[allow(clippy::not_unsafe_ptr_arg_deref)]
-pub fn load_string(rdb: *mut RedisModuleIO) -> String {
-    let p = unsafe { RedisModule_LoadString.unwrap()(rdb) };
-    RedisString::from_ptr(p)
+pub fn load_string(rdb: *mut RedisModuleIO) -> Result<String, Error> {
+    let p = unsafe { load::<*mut RedisModuleString>(rdb, RedisModule_LoadString)? };
```

Comment:
Same question as above

Reply:
Same answer (use of mutable static is unsafe)